search/src/Components/wrapper/reactive_search.js

```diff
@@ -557,11 +558,41 @@ class ReactiveSearchComponent extends Component {
         return (
             <ReactiveBase
                 app="weedid"
+                mapKey=""
```

Comment:
hmm... seeing as the key is available on the client side, I suppose there is other security around its use. In which case it would not be harmful to store the key in the repo.

In any case the key needs to be configurable from docker build (and should be documented in the README), so it makes sense to have it in `.env`?

Reply:
Does  [this](https://www.freecodecamp.org/news/how-to-implement-runtime-environment-variables-with-create-react-app-docker-and-nginx-7f9d42a91d70/) seem like a reasonable solution? Or it's too overkilling.

Reply:
I don't think we can safely copy all .env into a .js since it might include secrets.

It seems `process.env` can be used to include env variables from build time, and the challenge that that blog post attempts to solve is to make the values available at runtime. I don't really get why it does the string processing in bash rather than javascript. It seems to be the _right_ thing to do. Do it if you think it's worthwhile.

Although thinking a bit further: it's really just putting the variable value into a public file. We could do that other ways, too. Or deliver such config values from django.

And yet as I said, it seems to not be a very secure token; like a google analytics token, the security is not in keeping the API key secret.

Reply:
It seems we have to reveal it and restrict it from google console to assign valid domain. check [this](https://stackoverflow.com/a/39625963)